crm_compassion/models/weekly_demand.py

```diff
@@ -136,6 +154,34 @@ def _default_unsponsored_ambassador(self):
             'crm_compassion.number_children_ambassador'))
         return allocate_per_week - (float(ambass_sponsored) / 52.0)
 
+    @api.multi
+    def _compute_resupply_sub(self):
+        """ Compute SUB resupply. """
+        subs = list()
+        average_sub = self._compute_demand_sub()
+        sub_sponsorships = self.env['recurring.contract'].search([
+            ('origin_id.type', '=', 'sub'),
```

Comment:
You should add in the conditions that the start_date is 90 days before start_date of the week